soc/arm/renesas_rcar/gen4/soc.c

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2023 IoT.bzh
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+
+void z_arm_platform_init(void)
+{
+	L1C_DisableCaches();
+	L1C_DisableBTAC();
+
+	/* Invalidate instruction cache and flush branch target cache */
+	__set_ICIALLU(0);
+	__DSB();
+	__ISB();
```

Comment:
use the barriers API, see `include/zephyr/sys/barrier.h`

Reply:
Thanks, fixed